### PR TITLE
DSD-1520: Add sizeBasedOn prop to Logo

### PR DESCRIPTION
**Pull request by jackiequach**
Fixes [JIRA ticket 1520](https://jira.nypl.org/browse/DSD-1520)

## This PR does the following:

- Adds sizeBasedOn prop to Logo and corresponding info in Storybook docs
- Adds snapshot test for size based on height

## How has this been tested?

Locally in Storybook

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Review by EdwinGuzman — COMMENTED** [review]
The code looks good. I was trying to see if the logic in `logo.ts` could be done so that there's one object since the pixel values are the same for both `size` and `sizeBasedOn`. But the maxWidth and maxHeight does make that a bit difficult, and also the largest two sizes.

One thing that this is missing is the new component changelog table. Here's an example implementation: https://github.com/NYPL/nypl-design-system/blob/development/src/components/Slider/Slider.mdx#L339
and an example data file: https://github.com/NYPL/nypl-design-system/blob/development/src/components/Slider/sliderChangelogData.ts

**Review by EdwinGuzman — APPROVED** [review]
Looks good! Thanks for the last update. Let's also wait for Marty to review because he knows the styles better than I do.

File: src/components/Logo/Logo.mdx [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-| Component Version | DS Version |
-| ----------------- | ---------- |
-| Added             | `0.25.9`   |
-| Latest            | `1.5.2`    |
+| Component Version | DS Version   |
+| ----------------- | ------------ |
+| Added             | `0.25.9`     |
+| Latest            | `Prerelease` |
 
 ## Table of Contents
```
Comment (bigfishdesign13) [inline src/components/Logo/Logo.mdx:16]: Make sure to add the `Changelog` link to the TOC.

**Review by bigfishdesign13 — APPROVED** [review]
Works great!  Just make sure to update the TOC on the story page.

**Pull request merged**